src/main/java/util/JsonFileConverter.java

```diff
@@ -0,0 +1,89 @@
+package util;
+
+import com.google.gson.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Optional;
+
+/**
+ * Converts JSON file from and to object
+ */
+public class JsonFileConverter {
+    private static final Logger logger = LogManager.getLogger(JsonFileConverter.class.getName());
+    private static final String CHARSET = "UTF-8";
+
+    private Gson gson;
+    private final File jsonFile;
+
+    public JsonFileConverter(File jsonFile) {
+        this.jsonFile = jsonFile;
+        setupGson();
+        createDirectories();
+    }
+
+    /**
+     * Save a given object into JSON file
+     * @param object object to be saved to JSON file
+     */
+    public void saveToFile(Object object) {
+        try {
+            Writer writer = new OutputStreamWriter(new FileOutputStream(jsonFile) , CHARSET);
+            gson.toJson(object, object.getClass(), writer);
+            writer.close();
+        } catch (IOException e) {
```

Comment:
Not sure if the exception should be swallowed here. In general, if this fails to save, the caller should be allowed to react in some way.